components/brave_wallet/browser/eth_data_builder.cc

```diff
@@ -125,6 +125,71 @@ bool OwnerOf(uint256_t token_id, std::string* data) {
 
 }  // namespace erc721
 
+namespace erc1155 {
+bool SafeTransferFrom(const std::string& from,
+                      const std::string& to,
+                      uint256_t token_id,
+                      uint256_t value,
+                      std::string* data) {
+  const std::string function_hash = GetFunctionHash(
+      "safeTransferFrom(address,address,uint256,uint256,bytes)");
+
+  std::string padded_from;
+  if (!PadHexEncodedParameter(from, &padded_from)) {
+    return false;
+  }
+
+  std::string padded_to;
+  if (!PadHexEncodedParameter(to, &padded_to)) {
+    return false;
+  }
+
+  std::string padded_token_id;
+  if (!PadHexEncodedParameter(Uint256ValueToHex(token_id), &padded_token_id)) {
+    return false;
+  }
+
+  std::string padded_value;
+  if (!PadHexEncodedParameter(Uint256ValueToHex(value), &padded_value)) {
+    return false;
+  }
+
+  // SafeTransferFrom's `data` parameter is arbitrary bytes that the ERC1155
+  // contract will send as part of a onERC1155Received call if the recipient
+  // is an contract that implements ERC1155TokenReceiver.
+  // https://eips.ethereum.org/EIPS/eip-1155#erc-1155-token-receiver
+  //
+  // This argument is hardcoded as empty bytes since to support basic transfers
+  // only.
+  const std::string receiver_args =
+      "0x00000000000000000000000000000000000000000000000000000000000000a0000000"
+      "0000000000000000000000000000000000000000000000000000000000";
```

Comment:
Could you share more about how this value is generated? I thought it's 0x0 or so for empty bytes.

Reply:
0x0 for the bytes, but we also specify where those bytes "start" within the calldata, that's where 0xa0 comes in.  I added this to the comment:

```
The receiver_data_arg is hardcoded as empty bytes to support basic transfers
only.  It consists of two 32 byte parts. The first 32 specify the position of 
of the start of the bytes data in the SafeTransferFrom calldata:

0x00000000000000000000000000000000000000000000000000000000000000a0

This is interpreted as 'the data starts 0xa0 (160) bytes from the start of
the call data.' Since the preceding four arguments in the calldata
(to, from, id, amount) are all of fixed size (32 bytes), we can always
specify 0xa0 (160), since 32*(4+1) = 160.


The second 32 bytes are all 0s (the empty bytes):

0x0000000000000000000000000000000000000000000000000000000000000000
```

You take a look at this example 1155 safeTransferFrom that doesn't supply any data https://etherscan.io/tx/0xde123aaf42587723fe9fccfaf1772c71c2168b55b7efd62a57f30c074b37dc86 (go to 'Click to see More' then toggle 'Decode input data').

Reply:
Updated.  Let me know if I left too much detail in the comment.